js-arrays/arrays.js

```diff
@@ -12,24 +12,21 @@ function isArrayThree(a) {
 
 function range(a, b, c) {
   var arr = [];
-  if (b === undefined || b === null) {
-    if (c !== undefined) {
-      for (var i = 0; i < a; i += c) {
-        arr.push(i);
-      }
-    } else {
-      for (var i = 0; i < a; i++) {
-        arr.push(i);
-      }
-    }
-  } else if (c === undefined) {
-    for (var i = a; i < b; i++) {
-      arr.push(i);
-    }
-  } else if (c !== 0) {
-    for (var i = a; i < b; i += c) {
-      arr.push(i);
-    }
+  var i = 0;
+  var j, k;
+  if (b == undefined) {
+    j = a;
+  } else {
+    i = a;
+    j = b;
+  }
+  if (c == undefined || c <= 0) {
```

Comment:
looks much better! it can be improved to work with negative steps
_.range(-10, -20, -5) should return [-10, -15]
_.range(-20, -10, -5) should return []
I think another cases are ok